pkg/controller/gitopsconfig/controller.go

```diff
@@ -389,79 +392,110 @@ func (r *Reconciler) manageDeletion(instance *gitopsv1alpha1.GitOpsConfig) (reco
 	if !containsString(instance.ObjectMeta.Finalizers, tagFinalizer) {
 		return reconcile.Result{}, nil
 	}
-	// we need to lookup the delete job and if it doesn't exist we launch it, then we see if it is completed successfully if yes we remove the finalizers, if no we return.
-	jobList := &batchv1.JobList{}
-	selector, err := labels.Parse("action=delete")
+
+	// To avoid a deadlock situation let's check if the namespace in which we are is maybe being deleted?
+	ns := &corev1.Namespace{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{
+		Name: instance.GetNamespace(),
+	}, ns)
 	if err != nil {
-		log.Error(err, "unable to parse label selector 'action=delete'")
-		return reconcile.Result{}, xerrors.Errorf("unable to parse label selector 'action=delete': %w", err)
+		log.Error(err, "GitOpsConfig finalizer unable to lookup instance's namespace", "instance", instance.Name)
+		return reconcile.Result{}, xerrors.Errorf("GitOpsConfig finalizer unable to lookup instance's namespace for %q: %w", instance.Name, err)
 	}
-	// looking up all delete jobs
-	err = r.client.List(context.TODO(), &client.ListOptions{
-		Namespace:     instance.GetNamespace(),
-		LabelSelector: selector,
-	}, jobList)
+	if !ns.DeletionTimestamp.IsZero() {
+		//namespace is being deleted
```

Comment:
Out of curiosity, in most comments there is space after `//`. Is there any specific reason why in few places it's missing? I'll also mark other code lines.

Reply:
They're moved verbatim from fragments of old code, so I wasn't sure whether to change them or not; but maybe it makes sense to improve them in process indeed, thanks!